pandas/core/dtypes/missing.py

```diff
@@ -210,7 +210,7 @@ def _isna_ndarraylike(obj, inf_as_na: bool = False):
 
     if is_extension_array_dtype(dtype):
         if inf_as_na:
-            result = values.isna() | (values == -np.inf) | (values == np.inf)
+            result = libmissing.isnaobj_old(values.to_numpy())
```

Comment:
can we short-circuit this (and avoid the object-cast of to_numpy) for e.g. PeriodArray?

Reply:
Not in general, but we could short-circuit the known built-in dtypes that will never be able to contain inf (basically all our internal EA dtypes except for categorical ?)